### PR TITLE
Update change log for 3.1.0

### PR DESCRIPTION
**Pull request by TeBoring**

### CODE REVIEW

**Comment by TeBoring** [comment]
@pherl

File: CHANGES.txt [hunk 1]
```diff
@@ -1,3 +1,65 @@
+2016-09-23 version 3.1.0 (C++/Java/Python/PHP/Objective-C)
```
Comment (xfxyjwf) [inline CHANGES.txt]: Include all languages here.

File: CHANGES.txt [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+  General
+  * Proto3 support in PHP.
```
Comment (xfxyjwf) [inline CHANGES.txt]: Say explicitly it's alpha

File: CHANGES.txt [hunk 3]
```diff
@@ -1,3 +1,65 @@
...
+  * Various bug fixes.
+
+  C++
```
Comment (xfxyjwf) [inline CHANGES.txt]: Please incorporate haberman@'s comment into this change log:
https://github.com/google/protobuf/pull/2153

Comment (pherl) [inline CHANGES.txt]: Also there are performance optimizations to avoid static allocation. Gerben can probably add something for that.

Comment (TeBoring) [inline CHANGES.txt]: Done

Comment (TeBoring) [inline CHANGES.txt]: Done.

File: CHANGES.txt (hunk 2 above)
Comment (TeBoring) [inline CHANGES.txt]: Done

File: CHANGES.txt (hunk 1 above)
Comment (TeBoring) [inline CHANGES.txt]: Done

**Comment by pherl** [comment]
LGTM

File: CHANGES.txt [hunk 4]
```diff
@@ -1,3 +1,83 @@
...
+  * Fix #1342 JSON serialization of bytes not working.
+
+  Java
+  * Fixed a bug in TextFormat that doesn’t accept empty repeated fields (i.e.,
+    “field: [ ]”).
+  * JSON support
+    * Fixed JsonFormat to do correct snake_case-to-camelCase conversion for
+      non-style-conforming field names.
+    * Fixed JsonFormat to parse empty Any message correctly.
+    * Allow the JsonFormat.Parser to ignore unknown fields.
```
Comment (xfxyjwf) [inline CHANGES.txt]: Added an option to JsonFormat.Parser to ignore unknown fields.

File: CHANGES.txt [hunk 5]
```diff
@@ -1,3 +1,83 @@
...
+    MessageLite::ByteSize() but returns the value in size_t. Useful to check
+    whether a message is over the 2G size limit that protobuf can support.
+  * Moved default_instances to global variables. This allows default_instance
+    addresses to be known at compile time.
+  * Adding missing generic gcc 64-bit atomicops.
+  * Restore New*Callback into google::protobuf namespace since these are used
+    by the service stubs code
+  * JSON support.
+    * Fixed some conformance issues.
+  * Fix #1342 JSON serialization of bytes not working.
```
Comment (xfxyjwf) [inline CHANGES.txt]: Fixed a JSON serialization bug for bytes fields.

File: CHANGES.txt [hunk 6]
```diff
@@ -1,3 +1,83 @@
...
+    https://developers.google.com/protocol-buffers/docs/phptutorial for more
+    details about APIs.
+
+  Objective-C
+  * Helpers are now provided for working the the Any well known type (see
+    GPBWellKnownTypes.h for the api additions).
+  * Some improvements in startup code (especially when extensions aren’t used).
+
+  Javascript
+  * Fix missing import of jspb.Map
```
Comment (xfxyjwf) [inline CHANGES.txt]: Fix => Fixed

File: CHANGES.txt (hunk 4 above)
Comment (TeBoring) [inline CHANGES.txt]: Done.

File: CHANGES.txt (hunk 5 above)
Comment (TeBoring) [inline CHANGES.txt]: Done.

File: CHANGES.txt (hunk 6 above)
Comment (TeBoring) [inline CHANGES.txt]: Done.

File: CHANGES.txt [hunk 7]
```diff
@@ -1,3 +1,83 @@
...
+  Javascript
+  * Fixed missing import of jspb.Map
+  * Fixed valueWriterFn variable name
+
+  Ruby
+  * Fixed hash computation for JRuby's RubyMessage
+  * Make sure map parsing frames are GC-rooted.
+  * Added API support for well-known types.
+
+  Csharp
```
Comment (xfxyjwf) [inline CHANGES.txt]: C#

**Pull request merged**